planner/core/planbuilder.go

```diff
@@ -3081,6 +3081,10 @@ func (b *PlanBuilder) buildSimple(ctx context.Context, node ast.StmtNode) (Plan,
 					b.visitInfo = appendDynamicVisitInfo(b.visitInfo, "CONNECTION_ADMIN", false, err)
 					b.visitInfo = appendVisitInfoIsRestrictedUser(b.visitInfo, b.ctx, &auth.UserIdentity{Username: pi.User, Hostname: pi.Host}, "RESTRICTED_CONNECTION_ADMIN")
 				}
+			} else if raw.ConnectionID == util2.GetAutoAnalyzeProcID() {
+				// Only the users with the SUPER privilege can kill auto analyze.
+				err := ErrSpecificAccessDenied.GenWithStackByArgs("SUPER")
+				b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SuperPriv, "", "", "", err)
```

Comment:
Historically `PROCESS` privilege is required to kill connections that belong to other users. MySQL 8.0 changes this to requiring `CONNECTION_ADMIN`: https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_connection-admin

This helps avoid the problem that super is overloaded (it is used by too many features, which makes it less secure).

Reply:
Fixed.